privacy_analytics/managers.py

```diff
@@ -0,0 +1,25 @@
+from django.db.models import Manager, Avg, Count, Q, FloatField
+from django.db.models.functions import Cast
+
+
+class PageViewManager(Manager):
+    def unique(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').distinct().count()
+
+    def views_per_visitor(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').annotate(total=Count('pk')).aggregate(Avg('total'))
+
+    def total_views(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').count()
+
+    def percent_authenticated(self, **kwargs):
+        return self.filter(**kwargs).aggregate(total=(Cast(Count('is_authenticated', filter=Q(is_authenticated=True)), FloatField()) / Cast(Count('pk'), FloatField())) * 100.00)
+    
+    def referrers(self, **kwargs):
+        return self.filter(**kwargs).values('referrer').annotate(total=Count('pk')).all()[:10]
+    
+    def pages(self, **kwargs):
+        return self.filter(**kwargs).values('path').annotate(total=Count('pk'), percent=(Count('pk') * 100) / self.filter(**kwargs).all().count()).order_by('-total')
+    
+    def clear_by_time(self, **kwargs):
```

Comment:
What's the difference between calling `clear_by_time(...)` and `filter(...).delete()` (where `...` is the same)? Think I might be missing something here — what is the value added here?

Reply:
So with this, I pass in kwargs from the forms in the DashboardView, they are in the form of `time__lte` or `time__gte`, etc., (thinking about adding customising options here for more complex usage).

So we get all the PageView objects filtered by our time constraints (if any) and then remove based off of that.